consensus/consensus_algorithm.go

```diff
@@ -0,0 +1,69 @@
+package consensus
+
+import (
+	"github.com/spacemeshos/go-spacemesh/consensus/pb"
+	"time"
+)
+
+// OpaqueMessage is a basic blob of bytes type interface
+type OpaqueMessage interface {
+	// Data returns the bytes contained by this message
+	Data() []byte
+}
+
+// Algorithm is the main API to run a consensus algorithm that coordinates messages between nodes
+type Algorithm interface {
```

Comment:
You should define the interfaces when using them, best practice is not to have "global" insterfaces

Reply:
It helped me define what to test during tdd, lets discuss this